controllers/nonadmin_controller.go

```diff
@@ -177,10 +178,11 @@ func ensureRequiredSpecs(deploymentObject *appsv1.Deployment, image string) {
 
 func (r *DPAReconciler) checkNonAdminEnabled(dpa *oadpv1alpha1.DataProtectionApplication) bool {
 	// TODO https://github.com/openshift/oadp-operator/pull/1316
-	if dpa.Spec.Features != nil &&
-		dpa.Spec.Features.NonAdmin != nil &&
-		dpa.Spec.Features.NonAdmin.Enable != nil {
-		return *dpa.Spec.Features.NonAdmin.Enable
+	if dpa.Spec.NonAdmin != nil &&
+		dpa.Spec.NonAdmin.Enable != nil &&
+		boolptr.IsSetToTrue(dpa.Spec.NonAdmin.Enable) &&
```

Comment:
I would not add validation of unsupported override here, it is already checked by validator https://github.com/openshift/oadp-operator/blob/fc6afbf2920123fb4edb573a20561ed40632bca5/controllers/dpa_controller.go#L94

Reply:
no, that is just user feedback info

this is what actually checks if it is enabled

would try to simplify to
```go
	if dpa.Spec.Features != nil &&
		dpa.Spec.Features.NonAdmin != nil &&
		dpa.Spec.Features.NonAdmin.Enable != nil {
		return dpa.Spec.UnsupportedOverrides[oadpv1alpha1.TechPreviewAck] == TrueVal && *dpa.Spec.Features.NonAdmin.Enable
```